src/utils/in-memory-cache.ts

```diff
@@ -0,0 +1,52 @@
+import { idb } from "@/common/db";
+
+const PACMAN_WALLETS = "pacman-wallets";
+
+const inMemoryCache = () => {
+  const cache: Map<string, unknown> = new Map<string, unknown>();
+  return {
+    get: function (key: string): any {
+      return cache.get(key);
+    },
+    set: function (key: string, val: unknown) {
+      cache.set(key, val);
+    },
+  };
+};
+
+export const cache = inMemoryCache();
+
+export const updateWalletCache = async (address: string) => {
+  await saveWallet(address);
+  const walletExists: string[] = cache.get(PACMAN_WALLETS);
+  if (walletExists) {
+    walletExists.push(address);
+    cache.set(PACMAN_WALLETS, walletExists);
+  } else {
+    cache.set(PACMAN_WALLETS, [address]);
+  }
+};
+
+export const getCacheWallets = async (): Promise<string[]> => {
+  let wallets: string[] = cache.get(PACMAN_WALLETS);
+  if (!wallets) {
+    wallets = await idb.many("select address from pacman_wallets");
+  }
+  return wallets ?? [];
+};
+
```

Comment:
We need to have a method to quickly check whether a wallet address is a tracked wallet

```suggestion
export const isCachedWallet = async (address: string) => {
  const cachedWallets: Record<string, boolean> = cache.get(PACMAN_WALLETS);
  
  if (!cachedWallets || !cachedWallets[address]) return false;
  
  return true;
};
```